common/Root.tsx

```diff
@@ -46,22 +46,43 @@ export default class Root extends Component<Props, State> {
       return <ErrorScreen error={error} />;
     }
 
-    // key={Math.random()} = hack for HMR from https://github.com/webpack/webpack-dev-server/issues/395
+    const RouteWithSubRoutes = ({ path, subRoutes, BaseComponent }) => (
+      <Route
+        path={path}
+        render={props => {
+          const validRoutes = [path, ...subRoutes].filter(
+            route => subRoutes.includes(route) || route === path
+          );
+          return validRoutes.includes(props.location.pathname) ? (
+            <BaseComponent {...props} />
+          ) : (
+            <PageNotFound {...props} />
+          );
+        }}
+      />
+    );
```

Comment:
While tedious, I'd rather we use the `exact={true}` prop because all of the other routes (Not just these sub-tabbed ones) have the same issue of rendering anything at `route/*`, and this doesn't fix them (e.g. /ens/blah).